### PR TITLE
feat: my Account menu navigation grouping

### PR DESCRIPTION
**Pull request by tdonthu**
## PR Type

[x] Feature

## What Is the Current Behavior?
Menu navigation in My Account page is currently displayed as a plain listing without any hierarchy/grouping.

## What Is the New Behavior?
Menu navigation in My Account page will display group wise for B2B similar to the Responsive Starter Store.
The plain listing is kept for the B2C theme.

image

## Does this PR Introduce a Breaking Change?

[x] Yes

Changed NavigationItem data structure and handling (see migration notes).

## Other Information

[AB#81337](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/81337)

### CODE REVIEW

**Comment by SGrueber** [comment]
It would be good if the first menu item is always expanded after the user logged in. This might help the user to see that there are more options in the menu than he can see at first glance.
If the menu is collapsed lines are missing between the navigation items.

**Comment by shauke** [comment]
@iwiederhold @schadii Please review the changed Account Navigation if it meets your ideas. Please consider and answer the issues Silke addressed in her review. Styling was already discussed, so the missing lines are currently intended, but could be changed if necessary. Especially the partitioning should probably be discussed. Thank you.

**Review by iwiederhold — DISMISSED** [review]

**Review by schadii — DISMISSED** [review]

**Comment by SGrueber** [comment]
The logout link has been moved from the top to the button. If the organization menu is open it looks like an organization item ?

image

**Comment by SGrueber** [comment]
Why is payment under MyOrganization. Aren't payment instruments, like credit cards etc. specific to the user?

File: src/app/pages/account/account-navigation/account-navigation.items.b2c.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import { NavigationItem } from './account-navigation.component';
+
+// not-dead-code
+export const navigationItems: NavigationItem[] = [
+  { id: 'my-account', localizationKey: 'account.my_account.link', routerLink: '/account' },
+  {
```
Comment (SGrueber) [inline src/app/pages/account/account-navigation/account-navigation.items.b2c.ts:6]: even if you switch on b2b features in the environment.properties they won't work in b2c - so imo it is not necessary to leave them in this file: requisitions, approvals, costcenters, user management

**Comment by SGrueber** [comment]
The My Organization menu item is empty for a punchout user and should not be visible.


image

**Review by SGrueber — CHANGES_REQUESTED** [review]

**Review by SGrueber — APPROVED** [review]

**Review by schadii — DISMISSED** [review]
Thanks.

**Review by iwiederhold — DISMISSED** [review]
👍

**Review by shauke — APPROVED** [review]

**Pull request merged**